go/ql/test/experimental/CWE-942/vendor/github.com/gin-contrib/cors/stub.go

```diff
@@ -0,0 +1,41 @@
+// Code generated by depstubber. DO NOT EDIT.
+// This is a simple stub for github.com/gin-contrib/cors, strictly for use in testing.
+
+// See the LICENSE file for information about the licensing of the original library.
+// Source: github.com/gin-contrib/cors (exports: Config; functions: New)
+
+// Package cors is a stub of github.com/gin-contrib/cors, generated by depstubber.
+package cors
+
+import (
+	time "time"
+)
+
+type Config struct {
+	AllowAllOrigins        bool
+	AllowOrigins           []string
+	AllowOriginFunc        func(string) bool
+	AllowMethods           []string
+	AllowHeaders           []string
+	AllowCredentials       bool
+	ExposeHeaders          []string
+	MaxAge                 time.Duration
+	AllowWildcard          bool
+	AllowBrowserExtensions bool
+	AllowWebSockets        bool
+	AllowFiles             bool
+}
+
+func (_ Config) Validate() error {
+	return nil
+}
+
+func (_ *Config) AddAllowHeaders(_ ...string) {}
+
+func (_ *Config) AddAllowMethods(_ ...string) {}
+
+func (_ *Config) AddExposeHeaders(_ ...string) {}
+
+func New(_ Config) interface{} {
```

Comment:
```suggestion
func New(_ Config) gin.HandlerFunc {
```
This will get rid of some compilation problems, which are caused by depstubber not creating good enough stubs. It seems easiest just to fix the stubs.